src/physics_common.cpp

```diff
@@ -9,17 +9,13 @@ namespace openmc {
 // RUSSIAN_ROULETTE
 //==============================================================================
 
-void russian_roulette(Particle& p)
+void russian_roulette(Particle& p, double weight_survive)
 {
-  if (p.wgt() < settings::weight_cutoff) {
-    if (prn(p.current_seed()) < p.wgt() / settings::weight_survive) {
-      p.wgt() = settings::weight_survive;
-      p.wgt_last() = p.wgt();
-    } else {
-      p.wgt() = 0.;
-      p.wgt_last() = 0.;
-      p.alive() = false;
-    }
+  if (weight_survive * prn(p.current_seed()) < p.wgt()) {
```

Comment:
You could collapse all these lines into a two:

```
p.wgt() = weight_survive * prn(p.current_seed()) < p.wgt() ? weight_survive : 0.0;
p.alive() = p.wgt() != 0.0;
```

Also, why do we even use `p.alive()` as a separate variable? I think it would make more sense to just redefine `alive()` as:

```
bool alive() const { return _wgt != 0.0; }
```

Since the purpose of this PR is to remove an unnecessary data member, it's perhaps worth considering.

Now stylistically, maybe the ternary is less readable. But adopting these changes, this can be reduced to only one line.

Reply:
That's a great idea! It should work but I'll check first if it causes any unanticipated problems.

Reply:
For sure!